crates/holochain/src/conductor/conductor.rs

```diff
@@ -710,19 +724,22 @@ mod dna_impls {
                 .write_async({
                     let zome_defs = zome_defs.clone();
                     move |txn| {
-                        for dna_wasm in wasms {
-                            if !holochain_state::wasm::contains(txn, dna_wasm.as_hash())? {
-                                holochain_state::wasm::put(txn, dna_wasm)?;
+                        for wasm in wasms {
+                            if !holochain_state::wasm::contains(txn, wasm.as_hash())? {
+                                holochain_state::wasm::put(txn, wasm)?;
                             }
                         }
 
                         for (key, entry_def) in zome_defs.clone() {
                             holochain_state::entry_def::put(txn, key, &entry_def)?;
                         }
 
-                        if !holochain_state::dna_def::contains(txn, dna.as_hash())? {
```

Comment:
New `DnaDef`s would have never been persisted at all due to this `if` statement so coordinator udpates couldn't possibly be persisted across conductor restarts.